app/scripts/lib/rpc-method-middleware/handlers/caip-25.ts

```diff
@@ -0,0 +1,143 @@
+// rpcEndpoints (optional) = An array of URLs that each dereference to an RPC endpoints for routing requests within this scope.
+// These are ordered from most authoritative to least, i.e. priority SHOULD be given to endpoints in the order given, as per the CAIP-211 profile for that namespace, if one has been specified.
+
+//     "eip155": {
+//       "scopes": ["eip155:1", "eip155:137"],
+//       "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "get_balance", "personal_sign"],
+//       "notifications": ["accountsChanged", "chainChanged"]
+//     },
+
+export type ScopeObject = {
+  scopes?: CaipChainId[]; // CaipChainId[]
+  methods: string[];
+  notifications: string[];
+  accounts?: string[]; // CaipAccountId
+  rpcDocuments?: string[];
+  rpcEndpoints?: string[];
+};
+
+// Make this an assert
+export const isValidScope = (
+  scopeString: string,
+  scopeObject: ScopeObject,
+): boolean => {
+  const isNamespaceScoped = isCaipNamespace(scopeString);
+  const isChainScoped = isCaipChainId(scopeString);
+
+  if (!isNamespaceScoped && !isChainScoped) {
+    return false;
+  }
+
+  const {
+    scopes,
+    methods,
+    notifications,
+    accounts,
+    rpcDocuments,
+    rpcEndpoints,
+    ...restScopeObject
+  } = scopeObject;
+
+  // These assume that the namespace has a notion of chainIds
+  if (isChainScoped && scopes) {
+    // TODO: Probably requires refactoring this helper a bit
+    // When a badly-formed request includes a chainId mismatched to scope
+    //   code = 5203
+    //   message = "Scope/chain mismatch"
+    // When a badly-formed request defines one chainId two ways
+    //  code = 5204
+    //  message = "ChainId defined in two different scopes"
+    return false;
+  }
+  if (isNamespaceScoped && scopes) {
+    const namespace = scopeString;
+    const areScopesValid = scopes.every((scope) => {
+      try {
+        return parseCaipChainId(scope).namespace === namespace;
+      } catch (e) {
+        // parsing caipChainId failed
+        console.log(e);
+        return false;
+      }
+    });
+
+    if (!areScopesValid) {
+      return false;
+    }
+  }
+
+  const areMethodsValid = methods.every(
+    (method) => typeof method === 'string' && method !== '',
+  );
+  if (!areMethodsValid) {
+    return false;
+  }
+
+  const areNotificationsValid = notifications.every(
+    (notification) => typeof notification === 'string' && notification !== '',
+  );
+  if (!areNotificationsValid) {
+    return false;
+  }
+
+  // not validating rpcDocuments or rpcEndpoints currently
+
+  // unexpected properties found on scopeObject
+  if (Object.keys(restScopeObject).length !== 0) {
+    return false;
+  }
+
+  return true;
+};
+
+// This doesn't belong here
+export const isSupportedNotification = (notification: string): boolean => {
+  return ['accountsChanged', 'chainChanged'].includes(notification);
+};
```

Comment:
we can also get notifications out of the openrpc doc via methods that do not have `result`. (we dont have any specced right now but in the future we could)